src/main/java/cloud/cholewa/game/logic/GameStarter.java

```diff
@@ -0,0 +1,41 @@
+package cloud.cholewa.game.logic;
+
+import cloud.cholewa.game.Player;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class GameStarter {
+
+    public static int setNumberOfPlayers(Scanner scanner) {
+        int number;
+
+        System.out.print("Podaj liczbę graczy (2-4): ");
+        number = scanner.nextInt();
+
+        if (number < 2 || number > 4) {
+            System.out.println("Błędna ilość graczy !");
+            System.exit(0);
+        }
+
+        return number;
+    }
+
+    public static List<Player> setPlayersName(Scanner scanner, int numberOfPlayers) {
+        List<Player> playerList = new ArrayList<>();
+
+        while (playerList.size() < numberOfPlayers) {
+            try {
+                System.out.print("Podaj imię gracza nr " + (playerList.size() + 1) + ": ");
+                String name = scanner.nextLine();
+                Player player = new Player(name);
+                playerList.add(player);
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage() + "\n");
```

Comment:
Generalnie to jest ok, ale można się zastanowić czy nie zrobić 'cenralnego' punktu obsługi wyjątków, tzn. w GameEngine dać wszystko co masz w dużego try-cacha i w nim łapać Illegal Arguemnt / State/ Mismatch exceptiony i tam wypisywać na ekran, a niech ta metoda (i pozostałe) zamiast obsługiwać wyjątek niech go rzucą do obsługi GameEnginowi

Ma to taką zaletę, że gdy będziesz chciał dodać np.wyjątków  logowanie to dodajesz w jednym miejscu,  a teraz mamy już chyba trzy różne miejsca gdzie rzucamy wyjątek - łapiemy i  wypisujemy po prostu message.

Takie łapanie wyjątków na poziomie metody ma sens jeśli biznesowo chcemy jakoś obsłużyć (np. jeśli ktoś przy podawaniu liczby graczy poda 7 - czyli za dużo - to łapiemy mismatch exception, logujemy i np. ustawiamy liczbę graczy na maksymalną ilość, czyli w naszym wypadkua na 4.

Reply:
Przeniosłem te wyjątki, które faktycznie nic nie robią, poza wyświetlaniem komunikatu. 
Jednak w jednym miejscu jest weryfikacja, czy wprowadzona wartość jest liczbą. Wyświetlany jest odpowiedni komunikat błędu, ale potem czyszczony jest scanner i oczekujemy wprowadzenia wartości ponownie.